packages/react-atlas-core/src/TextArea/TextArea.js

```diff
@@ -0,0 +1,156 @@
+import React, { cloneElement } from "react";
+import PropTypes from "prop-types";
+import { InputCore } from "../Input";
+import cx from "classnames";
+
+class TextArea extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    // Initial state
+    this.state = {
+      "value": props.value || "",
+      "remaining": props.maxLength,
+      "active": false,
+      "valid": true
+    };
+  }
+
+  _handleChange = event => {
+    event.persist();
+    let value = event.target.value;
+
+    if (this.props.maxLength) {
+      // Keep difference between maxlength and input value in state for count
+      this.setState({ "remaining": this.props.maxLength - value.length });
+    }
+
+    if (this.props.required) {
+      // Set valid state depending on InputCore state
+      this.setState({ "valid": this.inputRef.state.isValid });
+    }
+
+    if (this.props.onChange) {
+      // Execute app code
+      this.props.onChange(event);
+    }
+  };
+
+  _handleFocus = event => {
+    this.setState({ "active": true });
+  };
+
+  _handleBlur = event => {
+    this.setState({ "active": false });
+  };
+
+  render() {
+    const {
+      maxLength,
+      resizable,
+      small,
+      medium,
+      large,
+      disabled,
+      hidden,
+      ...props
+    } = this.props;
+
+    let remainingCount =
+      maxLength &&
+      <div styleName={cx("remainingCount")}>
+        {maxLength - this.state.remaining}/{maxLength}
+      </div>;
+
+    let wrapperClasses = cx(
+      {
+        hidden,
+        small,
+        medium,
+        large
+      },
+      "textareaWrapper"
+    );
+
+    let textAreaClasses = cx(
+      {
+        resizable,
+        disabled,
+        "active": this.state.active,
+        "invalid": !this.state.valid
+      },
+      "textarea"
+    );
+
+    return (
+      <div
+        styleName={wrapperClasses}
+        onFocus={this._handleFocus}
+        onBlur={this._handleBlur}
+      >
+        <InputCore
```

Comment:
Add label/header prop and required (*) styling. Also update default font styles.